der/tests/derive.rs

```diff
@@ -261,6 +339,60 @@ mod sequence {
     const ALGORITHM_IDENTIFIER_DER: &[u8] =
         &hex!("30 13 06 07 2a 86 48 ce 3d 02 01 06 08 2a 86 48 ce 3d 03 01 07");
 
+    #[test]
+    fn idp_test() {
+        let idp = IssuingDistributionPointExample::from_der(&hex!("30038101FF")).unwrap();
+        assert_eq!(idp.only_contains_user_certs, true);
+        assert_eq!(idp.only_contains_cacerts, false);
+        assert_eq!(idp.indirect_crl, false);
+        assert_eq!(idp.only_contains_attribute_certs, false);
+
+        let idp = IssuingDistributionPointExample::from_der(&hex!("30038201FF")).unwrap();
+        assert_eq!(idp.only_contains_user_certs, false);
+        assert_eq!(idp.only_contains_cacerts, true);
+        assert_eq!(idp.indirect_crl, false);
+        assert_eq!(idp.only_contains_attribute_certs, false);
+
+        let idp = IssuingDistributionPointExample::from_der(&hex!("30038401FF")).unwrap();
+        assert_eq!(idp.only_contains_user_certs, false);
+        assert_eq!(idp.only_contains_cacerts, false);
+        assert_eq!(idp.indirect_crl, true);
+        assert_eq!(idp.only_contains_attribute_certs, false);
+
+        let idp = IssuingDistributionPointExample::from_der(&hex!("30038501FF")).unwrap();
+        assert_eq!(idp.only_contains_user_certs, false);
+        assert_eq!(idp.only_contains_cacerts, false);
+        assert_eq!(idp.indirect_crl, false);
+        assert_eq!(idp.only_contains_attribute_certs, true);
+    }
+
+    // demonstrates default field that is not context specific
+    #[test]
+    fn extension_test() {
+        //  0  15: SEQUENCE {
+        //   2   3:   OBJECT IDENTIFIER basicConstraints (2 5 29 19)
+        //   7   1:   BOOLEAN TRUE
+        //  10   5:   OCTET STRING, encapsulates {
+        //  12   3:     SEQUENCE {
+        //  14   1:       BOOLEAN TRUE
+        //        :       }
+        //        :     }
+        //        :   }
+        let ext1 = ExtensionExample::from_der(&hex!("300F0603551D130101FF040530030101FF")).unwrap();
```

Comment:
One thing that might be fun with these is to use the comment support in `hex_literal`, e.g.:

```suggestion
        let ext1 = ExtensionExample::from_der(&hex!("
            300F // 15: SEQUENCE
            0603 // 3:   OBJECT IDENTIFIER basicConstraints (2 5 29 19)
            551D130101FF040530030101FF // etc
        ")).unwrap();
```